internal/appsec/ruleset_builder.go

```diff
@@ -27,13 +27,14 @@ type (
 	}
 	// rulesFragment can represent a full ruleset or a fragment of it.
 	rulesFragment struct {
-		Version    string               `json:"version,omitempty"`
-		Metadata   interface{}          `json:"metadata,omitempty"`
-		Rules      []ruleEntry          `json:"rules,omitempty"`
-		Overrides  []rulesOverrideEntry `json:"rules_override,omitempty"`
-		Exclusions []exclusionEntry     `json:"exclusions,omitempty"`
-		RulesData  []ruleDataEntry      `json:"rules_data,omitempty"`
-		Actions    []interface{}        `json:"actions,omitempty"`
+		Version     string               `json:"version,omitempty"`
+		Metadata    interface{}          `json:"metadata,omitempty"`
+		Rules       []ruleEntry          `json:"rules,omitempty"`
+		Overrides   []rulesOverrideEntry `json:"rules_override,omitempty"`
+		Exclusions  []exclusionEntry     `json:"exclusions,omitempty"`
+		RulesData   []ruleDataEntry      `json:"rules_data,omitempty"`
+		Actions     []interface{}        `json:"actions,omitempty"`
+		CustomRules []ruleEntry          `json:"custom_rules,omitempty"`
```

Comment:
WDYT of `[]interface{}` instead?

Reply:
No real strong opinion on this so it's fine by me